BackEnd/source/server.js

```diff
@@ -0,0 +1,257 @@
+}
+
+function checkLeftDiagonalWin(colAndRowIndex, gameBoardArray) { // PURE FUNCTION
+  let fourInARowCount = 0;
+
+  let columnPos = colAndRowIndex[0];
+  let rowPos = colAndRowIndex[1];
+  while (columnPos >= 3 && rowPos >= 3) {
+    fourInARowCount = 0;
+    for (let i = 1; i < 4; i += 1) {
+      if (gameBoardArray[columnPos - i][rowPos - i] !== null) {
+        if (gameBoardArray[columnPos][rowPos] === gameBoardArray[columnPos - i][rowPos - i]) {
+          fourInARowCount += 1;
+        } else {
+          break;
+        }
+      } else {
+        break;
+      }
+    }
+    if (fourInARowCount >= 3) {
+      return gameBoardArray[columnPos][rowPos];
+    }
+    columnPos -= 1;
+    rowPos -= 1;
+  }
+  return null;
+}
+
+function getRightDiagnoalPoint(columnIndex, rowIndex, gameBoardArray) { // PURE FUNCTION
+  let columnPos = columnIndex;
+  let rowPos = rowIndex;
+  let lowestRightDiagonal;
+
+  let foundEnd = false;
+  while (foundEnd === false) {
+    if (columnPos + 1 > gameBoardArray.length - 1 || rowPos + 1 > gameBoardArray[0].length - 1) {
+      lowestRightDiagonal = [columnPos, rowPos];
+      foundEnd = true;
+    }
+    columnPos += 1;
+    rowPos += 1;
+  }
+  return lowestRightDiagonal;
+}
+
+function checkWinner(columnIndex, rowIndex) {
+  const lowestLeftDiagPoint = getLeftDiagnoalPoint(columnIndex, rowIndex, state.gameBoardValues);
+  const lowestRightDiagPoint = getRightDiagnoalPoint(columnIndex, rowIndex, state.gameBoardValues);
+
+  if (checkColumnWin(columnIndex, state.gameBoardValues) != null) {
+    return checkColumnWin(columnIndex, state.gameBoardValues);
+  } if (checkRowWin(rowIndex, state.gameBoardValues) != null) {
+    return checkRowWin(rowIndex, state.gameBoardValues);
+  } if (checkRightDiagonalWin(lowestLeftDiagPoint, state.gameBoardValues) != null) {
+    return checkRightDiagonalWin(lowestLeftDiagPoint, state.gameBoardValues);
+  } if (checkLeftDiagonalWin(lowestRightDiagPoint, state.gameBoardValues) != null) {
+    return checkLeftDiagonalWin(lowestRightDiagPoint, state.gameBoardValues);
+  }
+  return null;
+}
+
+app.use(express.static('./FrontEnd/source'));
+app.use(express.json());
+
+app.post('/setup', (req, res) => {
+  updateStateValues(req.body);
+  state.gameBoardValues = setupGameBoardValuesArray(state.gameBoardRows, state.gameBoardColumns);
+  res.json({
+    result: state,
+  });
+});
+
+app.post('/placeCounter', (req, res) => {
+  res.json({
+    updatedState: state,
+    placedCounterPos: placeCounter(req.body.column),
+  });
+});
+
+app.post('/checkWin', (req, res) => {
+  const isWin = checkWinner(parseInt(req.body.column, 10), parseInt(req.body.row, 10));
+  res.json({
+    winDetected: isWin,
+  });
+});
+
+app.post('/resetBoard', (req, res) => {
+  updateStateValues(req.body);
+  state.gameBoardValues = resetBoardArray(state.gameBoardArray, state.gameBoardColumns, state.gameBoardValues);
+  res.json({
+    updatedState: state,
+  });
+});
+
+// app.get('/setup', (req, res) => {
```

Comment:
In a real project I'd expect unused example code like this to be removed before code was merged into master.